asyncio/selector_events.py

```
@@ -39,6 +39,17 @@ def _test_selector_event(selector, fd, event):
         return bool(key.events & event)
 
 
+if hasattr(socket, 'TCP_NODELAY'):
+    def _set_nodelay(sock):
+        try:
+            sock.setsockopt(socket.IPPROTO_TCP, socket.TCP_NODELAY, 1)
+        except OSError:
+            pass
```

Without extensive googling no. Golang ignores any errors when setting TCP_NODELAY for similar reasons.

It's an optional flag, raising an error doesn't make any sense here as it can break a perfectly working program in some obscure case. What I can do is to log it in debug mode, but I don't think we need that.